mantidproject/mantid
# Replace commonBoundaries by isCommonBins and CommonBinsValidator

`MatrixWorkspace::isCommonBins()` does the same job as `WorkspaceHelpers::commonBoundaries()` and caches the result potentially saving the full check. Where possible, the check was removed altogether and `CommonBinsValidator` was used instead. Where `CommonBinsValidator` was used to just check for common binning, `isCommonBins()` is now directly used.

**Report to:** [nobody].

**To test:**

Code review. Check that the usage of `CommonBinsValidator` makes sense and no leftover `#includes` can be found.

Fixes #24292.

*This does not require release notes* because this is an internal change and should not affect users except for some input validation which is now done earlier.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
